[IPET-2176] Fix for the issue reported by error: Exception message: Infinite loop detected

diff --git a/src/Rewards/Plugin/Tax/Model/Sales/Total/Quote/CommonTaxCollector/ShippingDataObject.php b/src/Rewards/Plugin/Tax/Model/Sales/Total/Quote/CommonTaxCollector/ShippingDataObject.php
index 50b1539..91dd440 100644
--- a/src/Rewards/Plugin/Tax/Model/Sales/Total/Quote/CommonTaxCollector/ShippingDataObject.php
+++ b/src/Rewards/Plugin/Tax/Model/Sales/Total/Quote/CommonTaxCollector/ShippingDataObject.php
@@ -40,13 +40,17 @@ class ShippingDataObject

     protected $taxConfig;

+    protected \Magento\Checkout\Model\Cart $customerCart;
+
     public function __construct(
         Config $config,
         PurchaseHelper $purchaseHelper,
         QuoteFactory $quoteFactory,
-        TaxConfig $taxConfig
+        TaxConfig $taxConfig,
+        \Magento\Checkout\Model\Cart $customerCart
     ) {
         $this->config         = $config;
+        $this->customerCart   = $customerCart;
         $this->purchaseHelper = $purchaseHelper;
         $this->quoteFactory   = $quoteFactory;
         $this->taxConfig      = $taxConfig;
@@ -77,7 +81,10 @@ class ShippingDataObject
         }

         $item = array_shift($items);
-        $quote = $item->getQuote();
+        // BEGIN PATCH
+        //$quote = $item->getQuote();
+        $quote = $this->customerCart->getQuote();
+        // END PATCH

         $purchase = $this->purchaseHelper->getByQuote($quote->getId());

--
2.30.1 (Apple Git-130)
